packages/store/src/services/backend-system.ts

```diff
@@ -81,5 +90,31 @@ export class SystemService implements Service<BackendSystem, BackendSystemKey> {
 }
 
 export function getInstance(logger: Logger, options: ServiceOptions = {}): SystemService {
+    if (!options.baseDirectory) {
+        ensureSettingsMigrated();
+        options.baseDirectory = getSapDevToolsDirectory();
+    }
     return new SystemService(logger, options);
 }
+
+/**
+ * Ensure settings are migrated from the old fiori tools directory to the new sap development tools directory.
+ */
+function ensureSettingsMigrated(): void {
+    const sapDevToolsDir = getSapDevToolsDirectory();
+    const migrationFlag = join(sapDevToolsDir, '.migrated');
+
+    if (existsSync(migrationFlag)) {
+        return;
+    }
+
+    const systemFileName = getEntityFileName(Entity.BackendSystem);
+    const legacyPath = join(getFioriToolsDirectory(), systemFileName);
+    const newPath = join(sapDevToolsDir, systemFileName);
+
+    if (existsSync(legacyPath)) {
+        mkdirSync(dirname(newPath), { recursive: true });
+        copyFileSync(legacyPath, newPath);
```

Comment:
What if the new file already exists?
Read it and merge content?

Reply:
I think it could happen if consumers of store module are released/ installed by customers out of sync. Which seems very possible IMO.

Reply:
Thanks Austin, I guess I thinking it could be a one time migration and we could avoid continious file reads/checks. But it makes sense - a consumer using an old version could create a new system and this would not be reflected in a different consumer using the latest.